Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -563,8 +563,8 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
         _ cleanup: ConnectionStateMachine.ConnectionAction.CleanUpContext,
         context: ChannelHandlerContext
     ) {
-        self.logger.debug("Cleaning up and closing connection.", metadata: [.error: "\(cleanup.error)"])
-        
+        self.logger.debug("Cleaning up and closing connection.", metadata: [.error: "\(String(reflecting: cleanup.error))"])
```

Comment:
Example of what it solves:

<kbd>
<img width="1012" alt="Screenshot 2024-01-01 at 9 20 09 PM" src="https://github.com/vapor/postgres-nio/assets/54685446/5d385ac8-2c5b-46b0-97b5-9045c57db165">
</kbd>